### ISSUE TITLE
What's the differences between the two rnn networks?

### ISSUE THREAD

**Issue by tttwwy** [fchollet/keras#721]
I have two kinds of rnn networks to generate sentence,just like a language model.
```py
model = Sequential()
model.add(SimpleRNN(2, 3, return_sequences=True))
model.add(Activation('time_distributed_softmax'))
model.compile(loss='categorical_crossentropy', optimizer='rmsprop')
```

```py
model = Sequential()
model.add(SimpleRNN(2, 3, return_sequences=False))
model.add(Activation('softmax'))
model.compile(loss='categorical_crossentropy', optimizer='rmsprop')
```
I am wondering what's the difference between the two rnn networks?

**Comment by elanmart** [comment]
1.) as far as I know, `time_distributed_softmax` is deprecated. Use `softmax` instead.
2.) the first network will output a sequence of hidden states `y_1 ... y_n`, while the second will output only the last state `y_n`.
From the docs:
>return_sequences: Boolean. Whether to return the last output in the output sequence, or the full sequence.

**Comment by tttwwy** [comment]
Does that means that the first network will consider all the output sequences when computing loss,but the second network will only consider the last output word when computing loss?

**Comment by fchollet** [comment]
Yes, that's it. The second network only returns the last output state, the first returns the *sequence* of successive output states.
